transport/tchannel/handler.go

```diff
@@ -72,18 +70,16 @@ type inboundCallResponse interface {
 	SetApplicationError() error
 }
 
-// responseWriter provides an interface similar to handlerWriter.
-//
-// It allows us to control handlerWriter during testing.
+// responseWriter enhances transport.ResponseWriter interface with transport specific
+// methods.
 type responseWriter interface {
```

Comment:
Struct that implements this interface was implicitly used as transport.ResponseWriter.
I'm making this obvious: implementation if used as transport.ResponseWriter in a first place. Other additional functions exist only for make it easier to write unit tests.

Reply:
AddHeaders, SetApplicationError and Write are part of the transport.ResponseWriter interface